src/components/pages/admin/contacts/RoleEditorItem.vue

```diff
@@ -0,0 +1,187 @@
+            />
+            <div class="flex gap-2">
+              <div>
+                <AppInput v-model="editRole.endDate" type="date" />
+              </div>
+              <div>
+                <AppInput v-model="editRole.endTime" type="time" />
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div class="my-3 flex py-3">
+          <AppButton type="submit" variant="primary" :loading="loading">{{
+            t('form.saveChanges')
+          }}</AppButton>
+          <AppButton variant="text" class="ml-2" @click="formVisible = false">{{
+            t('form.cancel')
+          }}</AppButton>
+        </div>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import AppLabel from '../../../../components/forms/AppLabel.vue';
+import AppInput from '../../../../components/forms/AppInput.vue';
+import AppButton from '../../../../components/forms/AppButton.vue';
+import AppRoundBadge from '../../../../components/AppRoundBadge.vue';
+import AppConfirmDialog from '../../../../components/AppConfirmDialog.vue';
+import AppRadioGroup from '../../../../components/forms/AppRadioGroup.vue';
+import {
+  MemberRoleData,
+  GetMemberData,
+} from '../../../../utils/api/api.interface';
+import { updateRole, deleteRole } from '../../../../utils/api/member';
+import { onBeforeMount, ref, reactive } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { formatLocale } from '../../../../utils/dates/locale-date-formats';
+
+const props = defineProps<{
+  role: MemberRoleData;
+  contact: GetMemberData;
+}>();
+const emit = defineEmits(['update']);
+const editRole = reactive({
+  role: '',
+  startDate: '',
+  startTime: '',
+  endDate: '',
+  endTime: '',
+});
+
+const { t } = useI18n();
+const formVisible = ref(false);
+const roleHasEndDate = ref(false);
+const showDeleteModal = ref(false);
+const loading = ref(false);
+
+async function handleFormSubmit() {
+  loading.value = true;
+  try {
+    await updateRole(props.contact.id, editRole.role, {
+      dateAdded: parseDateTime(editRole.startDate, editRole.startTime),
+      dateExpires: roleHasEndDate.value
+        ? parseDateTime(editRole.endDate, editRole.endTime)
+        : null,
+    });
+  } finally {
+    emit('update');
+    loading.value = false;
+    formVisible.value = false;
+  }
+}
+
+async function handleDeleteRole() {
+  try {
+    await deleteRole(props.contact.id, props.role.role);
+  } finally {
+    emit('update');
+    loading.value = false;
+    formVisible.value = false;
+  }
+}
+
+function isRoleCurrent(role: MemberRoleData): boolean {
+  const now = new Date();
+  return role.dateAdded < now && (!role.dateExpires || role.dateExpires > now);
+}
+
+function parseDateTime(date: string, time: string): Date {
+  return new Date(date + 'T' + time);
+}
+
+onBeforeMount(async () => {
+  loading.value = false;
+  editRole.role = props.role.role;
+  editRole.startDate = props.role.dateAdded.toISOString().split('T')[0];
```

Comment:
Would prefer using date formatting rather than string splitting, see [callout code](https://github.com/beabee-communityrm/beabee-frontend/blob/main/src/pages/admin/callouts/new.vue#L133)
```ts
editRole.startDate = format(props.role.dateAdded, 'yyyy-MM-dd');
editRole.startTime = format(props.roles.dateAdded, 'HH:mm');
```